Add --ignore-verification-warnings flag

Logiraptor opened the pull request:
- Ops Manager returns HTTP 207 in some cases to signal a verification warning
- This is different from a verification error, which is a 422

Relevant issue #335

pivotal-issuemaster commented:
@Logiraptor Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal-cf/om&pullRequestId=338)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal-cf/om&pullRequestId=338) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

Logiraptor commented:
Closing to fix CLA issue

Logiraptor commented:
Closing to fix CLA issue (this time with pivotal-cf org)

pivotal-issuemaster commented:
@Logiraptor Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal-cf/om&pullRequestId=338)!

kcboyle commented on acceptance/configure_director_test.go:
```diff
@@ -543,5 +548,43 @@ properties-configuration:
 	  }`))
 		})
+
+		It("configures the BOSH director using the API", func() {
```
What is unique about this test? You are specifically testing that this returns a 207, but I don't get this at all from your description

Logiraptor replied:
Agreed, that's an oversight on my part.

kcboyle commented on acceptance/help_test.go:
```diff
@@ -84,19 +85,20 @@ const CONFIGURE_AUTHENTICATION_USAGE = `ॐ  configure-authentication
+  --decryption-passphrase, -d, OM_DECRYPTION_PASSPHRASE    string  Passphrase to decrypt the installation if the Ops Manager VM has been rebooted (optional for most commands)
+  --env, -e                                                string  env file with login credentials
+  --help, -h                                               bool    prints this usage information (default: false)
+  --ignore-verifier-warnings, OM_IGNORE_VERIFIER_WARNINGS  bool    continues working even if there are verifier warnings present (default: false)
```
why is this a global option? `--ignore-verifier-warnings` should only be enabled for `configure-product` and `configure-director`.  A good place for this that would not affect other commands would be in the `Options` structs in `ConfigureDirector` and `ConfigureProduct`

Logiraptor replied:
I made the change to scope this flag to just configure-director, since that is the only known place where a 207 is returned.

kcboyle commented on main.go:
```diff
@@ -35,19 +35,20 @@ type httpClient interface {
-	Username             string `yaml:"username"              short:"u"  long:"username"            env:"OM_USERNAME"                            description:"admin username for the Ops Manager VM (not required for unauthenticated commands)"`
-	Env                  string `                             short:"e"  long:"env"                                                              description:"env file with login credentials"`
-	Version              bool   `                             short:"v"  long:"version"                                          default:"false" description:"prints the om release version"`
+	DecryptionPassphrase   string `yaml:"decryption-passphrase"    short:"d"  long:"decryption-passphrase"    env:"OM_DECRYPTION_PASSPHRASE"                    description:"Passphrase to decrypt the installation if the Ops Manager VM has been rebooted (optional for most commands)"`
```
changes that should only affect 2 commands should not change main.go

kcboyle commented on acceptance/help_test.go:
```diff
@@ -13,19 +13,20 @@ const GLOBAL_USAGE = `ॐ
+  --decryption-passphrase, -d, OM_DECRYPTION_PASSPHRASE    string  Passphrase to decrypt the installation if the Ops Manager VM has been rebooted (optional for most commands)
+  --env, -e                                                string  env file with login credentials
+  --help, -h                                               bool    prints this usage information (default: false)
+  --ignore-verifier-warnings, OM_IGNORE_VERIFIER_WARNINGS  bool    continues working even if there are verifier warnings present (default: false)
```
Can we make this description something more like: 
```
 option to ignore verifier warnings. NOT RECOMMENDED UNLESS DISABLED IN OPS MANAGER (default: false) 
```

The reason for this is that we do not want users to use this flag unless they absolutely know what they are doing. We want to protect people from themselves.

Logiraptor replied:
Sounds good.

Logiraptor commented:
@kcboyle Thanks for the review! I think I've addressed the feedback. 

The flag is now scoped very tightly to just the configure-director command, and internally only the call to the availability zones api. 

I believe this version should be good to go. Let me know if you have any other concerns.

Logiraptor commented:
Also, I notice the ci job is failing, but I don't have access to those logs. Let me know if there's something I can fix. All the tests pass for me locally.

kcboyle commented:
Hi @Logiraptor ! I believe the failure is an issue on our part with active development (we have pending tests, and this is not allowed). So if the tests pass locally, don't worry about it. We'll see if we can take another look today/tomorrow. Thanks for responding so quickly (and for the PR!)

kcboyle commented:
Merged via command line. Thanks for the PR!
